lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3281,6 +3281,17 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       // Record the body.
       TC.definedFunctions.push_back(FD);
     }
+    
+    // If any of the function's parameters contain an uninhabited type
+    // then emit a diagnostic on the func decl, as the function can never
+    // be called.
+    auto params = FD->getParameters()->getArray();
+    for (auto &&index: indices(params)) {
+      auto param = params[index];
+      if (param->hasInterfaceType() && param->getType()->isStructurallyUninhabited()) {
+        TC.diagnose(FD->getLoc(), diag::function_parameter_type_uninhabited, FD->getName(), param->getName());
```

Comment:
Lines should be < 80 characters

Reply:
Fixed. Now using the parameter's location instead of the function!